hathor/transaction/aux_pow.py

```diff
@@ -44,23 +41,26 @@ def calculate_hash(self, base_block_hash: bytes) -> bytes:
         merkle_root = bytes(reversed(build_merkle_root_from_path([coinbase_tx_hash] + self.merkle_path)))
         return sha256d_hash(self.header_head + merkle_root + self.header_tail)
 
-    def verify(self, _base_block_hash: bytes) -> None:
+    def verify(self, _base_block_hash: bytes, max_merkle_path_length: int) -> None:
         """ Check for inconsistencies, raises instance of TxValidationError on error.
         """
+        self.verify_magic_number(_base_block_hash)
+        self.verify_merkle_path(_base_block_hash, max_merkle_path_length)
+
+    def verify_magic_number(self, _base_block_hash: bytes) -> None:
         from hathor.merged_mining import MAGIC_NUMBER
-        from hathor.transaction.exceptions import (
-            AuxPowLongMerklePathError,
-            AuxPowNoMagicError,
-            AuxPowUnexpectedMagicError,
-        )
+        from hathor.transaction.exceptions import AuxPowNoMagicError, AuxPowUnexpectedMagicError
         magic_index = self.coinbase_head.find(MAGIC_NUMBER)
         if magic_index == -1:
             raise AuxPowNoMagicError('cannot find MAGIC_NUMBER')
         if magic_index < len(self.coinbase_head) - len(MAGIC_NUMBER):
             raise AuxPowUnexpectedMagicError('unexpected MAGIC_NUMBER')
+
+    def verify_merkle_path(self, _base_block_hash: bytes, max_merkle_path_length: int) -> None:
```

Comment:
Missing docstring.

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/910/commits/863db2aaa85a62ae1e6ea60dc5a5175868221da1

Reply:
Great! Thanks!